pkg/auth/jwt.go

```diff
@@ -0,0 +1,72 @@
+package auth
+
+import (
+	"fmt"
+	"os"
+	"net/http"
+	"context"
+
+	"NKSS-backend/pkg/handlers"
+
+	"github.com/golang-jwt/jwt"
+)
+
+func Authenticator(next http.Handler, secret []byte) http.HandlerFunc {
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Check authorization via JWT
+		tokenStr := r.Header.Get("Authorization")
+		rollno, ok := validateJWT(tokenStr, secret)
+		if tokenStr == "" || (rollno == "" && !ok) {
```

Comment:
just checking `ok` should sufficient
if it is not, then it should be a concern of `validateJWT`, not the caller
```suggestion
		if !ok {
```